src/components/EndorsementChain/EndorsementChainLayout/EndorsementChainLayout.tsx

```diff
@@ -28,35 +28,73 @@ const EndorsementChainLayoutUnstyled: FunctionComponent<EndorsementChainLayout>
-            </div>
-            <div className="table-cell endorsement-ui-dash">
-              {beneIndex === 0 && holderIndex === 0 && <div className="mask" />}
-              <AddressCell
-                address={beneficiaryChangeEvent.beneficiary}
-                titleEscrowAddress={beneficiaryChangeEvent.titleEscrowAddress}
-                newAddress={!(previousBeneficiary === beneficiaryChangeEvent.beneficiary)}
-              />
-            </div>
-            <div className="table-cell endorsement-ui-dash">
-              {beneIndex === 0 && holderIndex === 0 && <div className="mask" />}
-              <AddressCell
-                address={holderChangeEvent.holder}
-                titleEscrowAddress={beneficiaryChangeEvent.titleEscrowAddress}
-                newAddress={!(previousHolder === holderChangeEvent.holder)}
-              />
-            </div>
-          </div>
-        );
-        previousBeneficiary = beneficiaryChangeEvent.beneficiary;
-        previousHolder = holderChangeEvent.holder;
-      });
+    endorsementChain.forEach((tradetrustErc721Event, eventIndex) => {
+      switch (tradetrustErc721Event.eventType) {
+        case "Transfer":
+          const beneficiaryChangeEvent = tradetrustErc721Event as TitleEscrowEvent;
+          if (!beneficiaryChangeEvent.holderChangeEvents || !beneficiaryChangeEvent.beneficiary)
+            return new Error("Invalid Event: Transfer Event does not have new beneficiary or new holder address");
+          beneficiaryChangeEvent.holderChangeEvents.forEach((holderChangeEvent, holderIndex) => {
+            tableRows.push(
+              <tr className="table-row" key={index++}>
+                <td className="table-cell date">
+                  {format(new Date(holderChangeEvent.timestamp), "do MMM yyyy, hh:mm aa")}
+                </td>
+                <td className="table-cell endorsement-ui-dash">
+                  {eventIndex === 0 && holderIndex === 0 && <div className="mask" />}
+                  <AddressCell
+                    address={beneficiaryChangeEvent.beneficiary}
+                    titleEscrowAddress={beneficiaryChangeEvent.documentOwner}
+                    newAddress={!(previousBeneficiary === beneficiaryChangeEvent.beneficiary)}
+                  />
+                </td>
+                <td className="table-cell endorsement-ui-dash">
+                  {eventIndex === 0 && holderIndex === 0 && <div className="mask" />}
+                  <AddressCell
+                    address={holderChangeEvent.holder}
+                    titleEscrowAddress={beneficiaryChangeEvent.documentOwner}
+                    newAddress={!(previousHolder === holderChangeEvent.holder)}
+                  />
+                </td>
+              </tr>
+            );
+            previousBeneficiary = beneficiaryChangeEvent.beneficiary;
+            previousHolder = holderChangeEvent.holder;
+          });
+          return;
```

Comment:
```suggestion
          break;
```

Return will break if the forEach is change to a regular for loop, that's not what you intended i think.